tests/013-eventhandler.cpp

```diff
@@ -0,0 +1,148 @@
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/matchers/catch_matchers.hpp>
+#include <future>
+#include <oxen/quic.hpp>
+#include <oxen/quic/gnutls_crypto.hpp>
+#include <thread>
+
+#include "utils.hpp"
+
+namespace oxen::quic::test
+{
+    struct lifetime : public std::enable_shared_from_this<lifetime>
+    {};
+
+    TEST_CASE("013 - EventHandler event repeater: calling object lifetime bound", "[013][repeater][caller]")
+    {
+        Network test_net{};
+        const int NUM_ITERATIONS{10};
+        constexpr auto msg = "hello from the other siiiii-iiiiide"_bsv;
+
+        std::promise<bool> d_promise;
```

Comment:
The `bool` value that goes into this promise never varies and is never read; this should be a `std::promise<void>` instead.

Reply:
Actually, given where this is used below, this should stay a `promise<bool>` or maybe a `promise<std::pair<bool, bool>>`, and the conditions currently in the REQUIRE should get propagated via the promise's value to be checked via REQUIRE's outside the call_soon.

Reply:
See above response to the `REQUIRE` comment

Reply:
Not resolved: if this `bool` value is not being used, this should be a `promise<void>`.